cdp/test/test_cdp_parser.py

```diff
@@ -385,27 +385,31 @@ def test_get_parameters(self):
             p = self.cdp_parser.get_parameters(default_vars=False, cmd_default_vars=False, argparse_vals_only=False)[0]
             self.assertEqual(p.vars, ['v1'])
             self.cdp_parser.add_args_and_values(['-p', self.prefix + 'test_get_parameters.py', '-d', 'test_get_parameters.cfg', '--selectors', 'vars'])
-            p = self.cdp_parser.get_parameters(default_vars=False, cmd_default_vars=False, argparse_vals_only=False)[0]
-            self.assertEqual(p.vars, ['v2'])
+            p = self.cdp_parser.get_parameters(default_vars=False, cmd_default_vars=False, argparse_vals_only=False)
+            self.assertEqual(p, [])
 
             self.cdp_parser.add_args_and_values(['-p', self.prefix + 'test_get_parameters.py', '-v', 'v3'])
             p = self.cdp_parser.get_parameters(default_vars=False, cmd_default_vars=False, argparse_vals_only=False)[0]
             self.assertEqual(p.vars, ['v3'])
-            self.cdp_parser.add_args_and_values(['-p', self.prefix + 'test_get_parameters.py', '-v', 'v3', '--selectors', 'vars'])
+
+            self.cdp_parser.add_args_and_values(['-p', self.prefix + 'test_get_parameters.py', '--selectors', 'vars'])
             p = self.cdp_parser.get_parameters(default_vars=False, cmd_default_vars=False, argparse_vals_only=False)[0]
+            # Remember, selectors parameter only selects from the cfg (-d) files.
             self.assertEqual(p.vars, ['v1'])
 
             self.cdp_parser.add_args_and_values(['-d', 'test_get_parameters.cfg', '-v', 'v3'])
             p = self.cdp_parser.get_parameters(default_vars=False, cmd_default_vars=False, argparse_vals_only=False)[0]
             self.assertEqual(p.vars, ['v3'])
+
             self.cdp_parser.add_args_and_values(['-d', 'test_get_parameters.cfg', '-v', 'v3', '--selectors', 'vars'])
-            p = self.cdp_parser.get_parameters(default_vars=False, cmd_default_vars=False, argparse_vals_only=False)[0]
-            self.assertEqual(p.vars, ['v2'])
+            p = self.cdp_parser.get_parameters(default_vars=False, cmd_default_vars=False, argparse_vals_only=False)
```

Comment:
why do look at p == [] when it used to be p.vars == ['v2']

Reply:
In master, I made a change to line 404, which added the `selectors` argument.
```
self.cdp_parser.add_args_and_values(['-d', 'test_get_parameters.cfg', '-v', 'v3', '--selectors', 'vars'])
```
This is why it's not red. Should also be marked as red, so I'm not cheating.
Forgive that mistake.